### ISSUE TITLE
compile warnings with g++5.4.1

### ISSUE THREAD

**Issue by jsteemann** [iresearch-toolkit/iresearch#2]
When compiling everything with g++-5.4.1 (g++ (Ubuntu 5.4.1-2ubuntu1~16.04) 5.4.1 20160904) the compiler reports the following warnings:
```
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp: In instantiation of ‘void {anonymous}::__fastpack(const uint32_t*, uint32_t*) [with int N = 1; uint32_t = unsigned int]’:
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:553:36:   required from here
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:30:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  1) % 32; if (((N *  2) % 32) < ((N *  1) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  2) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:31:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  2) % 32; if (((N *  3) % 32) < ((N *  2) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  3) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:32:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  3) % 32; if (((N *  4) % 32) < ((N *  3) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  4) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:33:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  4) % 32; if (((N *  5) % 32) < ((N *  4) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  5) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:34:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  5) % 32; if (((N *  6) % 32) < ((N *  5) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  6) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:35:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  6) % 32; if (((N *  7) % 32) < ((N *  6) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  7) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:36:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  7) % 32; if (((N *  8) % 32) < ((N *  7) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  8) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:37:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  8) % 32; if (((N *  9) % 32) < ((N *  8) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N *  9) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:38:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N *  9) % 32; if (((N * 10) % 32) < ((N *  9) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 10) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:39:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 10) % 32; if (((N * 11) % 32) < ((N * 10) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 11) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:40:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 11) % 32; if (((N * 12) % 32) < ((N * 11) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 12) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:41:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 12) % 32; if (((N * 13) % 32) < ((N * 12) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 13) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:42:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 13) % 32; if (((N * 14) % 32) < ((N * 13) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 14) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:43:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 14) % 32; if (((N * 15) % 32) < ((N * 14) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 15) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:44:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 15) % 32; if (((N * 16) % 32) < ((N * 15) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 16) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:45:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 16) % 32; if (((N * 17) % 32) < ((N * 16) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 17) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:46:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 17) % 32; if (((N * 18) % 32) < ((N * 17) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 18) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:47:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 18) % 32; if (((N * 19) % 32) < ((N * 18) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 19) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:48:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 19) % 32; if (((N * 20) % 32) < ((N * 19) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 20) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:49:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 20) % 32; if (((N * 21) % 32) < ((N * 20) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 21) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:50:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 21) % 32; if (((N * 22) % 32) < ((N * 21) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 22) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:51:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 22) % 32; if (((N * 23) % 32) < ((N * 22) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 23) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:52:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 23) % 32; if (((N * 24) % 32) < ((N * 23) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 24) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:53:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 24) % 32; if (((N * 25) % 32) < ((N * 24) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 25) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:54:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 25) % 32; if (((N * 26) % 32) < ((N * 25) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 26) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:55:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 26) % 32; if (((N * 27) % 32) < ((N * 26) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 27) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:56:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 27) % 32; if (((N * 28) % 32) < ((N * 27) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 28) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:57:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 28) % 32; if (((N * 29) % 32) < ((N * 28) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 29) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:58:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 29) % 32; if (((N * 30) % 32) < ((N * 29) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 30) % 32)); }
                                                                                                                                   ^
/home/steemann/Downloads/iresearch-build/iresearch/core/utils/bit_packing.cpp:59:131: warning: right shift count is negative [-Wshift-count-negative]
   ++in; *out |= ((*in) % (1U << N)) << (N * 30) % 32; if (((N * 31) % 32) < ((N * 30) % 32)) { ++out; *out |= ((*in) % (1U << N)) >> (N - ((N * 31) % 32)); }
 ```

**Comment by gnusi** [comment]
Fixed in fb7c9aff0f3c75f52d65dd0a44c9c7b9dc58739a

**Comment by jsteemann** [comment]
Thanks, the warnings are gone now.
